internal/localdebug/local_ide_debug.go

```diff
@@ -0,0 +1,263 @@
+func findIdeaFolder(startPath string) string {
+	currentPath := startPath
+
+	for {
+		ideaPath := filepath.Join(currentPath, ".idea")
+		if _, err := os.Stat(ideaPath); err == nil {
+			return ideaPath
+		}
+
+		parentPath := filepath.Dir(currentPath)
+		if parentPath == currentPath {
+			// Reached the root directory
+			break
+		}
+		currentPath = parentPath
+	}
+
+	return ""
+}
+
+func handleVSCode(cxt context.Context, path string, ports map[string]*DebugInfo) {
+	logger := log.FromContext(cxt)
+	launchJSON := findLaunchJSON(path)
+	if launchJSON == "" {
+		return
+	}
+
+	contents := map[string]interface{}{}
+	existing := map[string]int{}
+	var configurations []interface{}
+	if _, err := os.Stat(launchJSON); err == nil {
+		file, err := os.ReadFile(launchJSON)
+		if err != nil {
+			logger.Errorf(err, "could not read launch.json")
+			return
+		}
+		err = json.Unmarshal(file, &contents)
+		if err != nil {
+			logger.Errorf(err, "could not read launch.json")
+			return
+		}
+		configurations = contents["configurations"].([]interface{}) //nolint:forcetypeassert
+		if configurations == nil {
+			configurations = []interface{}{}
+		}
+	} else {
+		contents["version"] = "0.2.0"
+		configurations = []interface{}{}
+	}
+	for i, config := range configurations {
+		existing[config.(map[string]interface{})["name"].(string)] = i //nolint:forcetypeassert
+	}
+
+	for k, v := range ports {
+		if v.Language == "java" || v.Language == "kotlin" {
+			name := "FT𝝺 JVM - " + k
+			pos, ok := existing[name]
+			if ok {
+				// Update the port
+				configurations[pos].(map[string]interface{})["port"] = v.Port //nolint:forcetypeassert
+				continue
+			}
+			entry := map[string]interface{}{
+				"name":     name,
+				"type":     "java",
+				"request":  "attach",
+				"hostName": "127.0.0.1",
+				"port":     v.Port,
+			}
+			configurations = append(configurations, entry)
+
+		} else if v.Language == "go" {
+			name := "FT𝝺 GO - " + k
+			pos, ok := existing[name]
+			if ok {
+				// Update the port
+				configurations[pos].(map[string]interface{})["port"] = v.Port //nolint:forcetypeassert
+				continue
+			}
+			entry := map[string]interface{}{
+				"name":       name,
+				"type":       "go",
+				"request":    "attach",
+				"mode":       "remote",
+				"apiVersion": 2,
+				"host":       "127.0.0.1",
+				"port":       v.Port,
+			}
+			configurations = append(configurations, entry)
+		}
+	}
+	contents["configurations"] = configurations
+	data, err := json.Marshal(contents)
+	if err != nil {
+		logger.Errorf(err, "could not marshal launch.json")
+		return
+	}
+	err = os.WriteFile(launchJSON, data, 0600)
+	if err != nil {
+		logger.Errorf(err, "could not write launch.json")
+		return
+	}
+}
+
+// findLaunchJSON recurses up the directory tree to find a .vscode folder
+// If it can't find one it creates one next to project.toml
+// It then returns the path to the launch.json file
+func findLaunchJSON(startPath string) string {
```

Comment:
Let's change this to terminate at the [project root](https://github.com/TBD54566975/ftl/blob/1dc16786a031b775965e209f28478f3117708370/internal/projectconfig/projectconfig.go#L28) rather than the root directory, so we don't randomly start writing into an unrelated VSCode directory. Same with the IDEA stuff.